FormalConjectures/Paper/HartshorneConjecture.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+import FormalConjectures.Util.ProblemImports
+
+/-! # Hartshorne's conjecture on Vector Bundles
+
+*Reference*: [Varieties of small codimension in projective space](https://projecteuclid.org/journals/bulletin-of-the-american-mathematical-society-new-series/volume-80/issue-6/Varieties-of-small-codimension-in-projective-space/bams/1183535999.full)
+by *R. Hartshorne*
+-/
+
+open CategoryTheory Limits MvPolynomial AlgebraicGeometry
+
+variable (S : Scheme)
+
+namespace AlgebraicGeometry.Scheme
+
+section AlgebraicVectorBundles
+
+attribute [local instance] CategoryTheory.Types.instConcreteCategory
+
+/--
+A vector bundle over a scheme `S` is a locally free `𝓞_S`-module of finite rank.
+-/
+structure VectorBundles where
+  carrier : S.Modules
+  rank : ℕ
+  isLocallyFreeFiniteConstantRank : SheafOfModules.IsVectorBundleWithRank
+    (J := Opens.grothendieckTopology S) carrier rank
+
+instance (S : Scheme) : Coe S.VectorBundles S.Modules where
+  coe 𝓕 := 𝓕.carrier
+
+/--
+Vector bundles form a category.
+-/
+instance : Category S.VectorBundles :=
+  InducedCategory.category VectorBundles.carrier
+
+def VectorBundles.toModule : S.VectorBundles ⥤ S.Modules where
+  obj 𝓕 := 𝓕.carrier
+  map f := f
+
+@[category API, AMS 14]
+theorem hasFiniteCoproductsVectorBundles : HasFiniteCoproducts S.VectorBundles := by
+  sorry
+
+instance : HasFiniteCoproducts S.VectorBundles :=
+  hasFiniteCoproductsVectorBundles S
```

Comment:
Why is the theorem one not an instance already? Is this for compatibility with the attributes tag?